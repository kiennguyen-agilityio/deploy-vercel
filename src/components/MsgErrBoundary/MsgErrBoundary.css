.error-fallback-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 200px;
  padding: 20px;
  background-color: var(--color-background-primary);
  border: 1px solid var(--color-border-primary);
  border-radius: 4px;
  font-size: var(--font-size-18);
  color: var(--color-button-hover);
}

.error-fallback-title {
  margin-bottom: 20px;
}

.error-fallback-description {
  margin-bottom: 20px;
  text-align: center;
}

.error-fallback-button {
  padding: 10px 20px;
  background-color: var(--color-background-secondary);
  color: var(--color-white);
  border: none;
  border-radius: 4px;
  cursor: pointer;
}
