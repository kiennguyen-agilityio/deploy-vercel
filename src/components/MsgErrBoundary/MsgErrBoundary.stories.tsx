import { StoryObj, Meta } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';

import MsgErrBoundary from './MsgErrBoundary';

export default {
  title: 'Components/MsgErrBoundary',
  component: MsgErrBoundary,
  decorators: [
    (Story) => (
      <MemoryRouter>
        <Story />
      </MemoryRouter>
    ),
  ],
} as Meta;

type Story = StoryObj<typeof MsgErrBoundary>;

export const Default: Story = {};
