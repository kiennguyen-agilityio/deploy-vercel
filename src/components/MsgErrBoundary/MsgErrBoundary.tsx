import { useState } from 'react';
import { useNavigate } from 'react-router-dom';

// constants
import ROUTE from '@/constants/route';
import { BUTTON_VARIANTS, BUTTON_SIZES, BUTTON_COLORS } from '@/enums/index';

// components
import Modal from '@/components/Modal/Modal';
import Button from '@/components/Button/Button';

// style
import './MsgErrBoundary.css';

const MsgErrBoundary: React.FC = () => {
  const [isOpen, setIsOpen] = useState<boolean>(true);
  const navigate = useNavigate();

  const handleCloseModal = (): void => {
    setIsOpen(false);
    navigate(ROUTE.HOMEPAGE);
  };

  return (
    <Modal isOpen={isOpen} onClose={handleCloseModal}>
      <div className="error-fallback-container">
        <h2 className="error-fallback-title">Oops! Something went wrong</h2>
        <p className="error-fallback-description">
          We&apos;re sorry, but an error occurred while processing your request.
        </p>
        <Button
          label="Back to home page"
          onClick={handleCloseModal}
          variant={BUTTON_VARIANTS.PRIMARY}
          isDisabled={false}
          isLoading={false}
          size={BUTTON_SIZES.SMALL}
          color={BUTTON_COLORS.DEFAULT}
        />
      </div>
    </Modal>
  );
};

export default MsgErrBoundary;
