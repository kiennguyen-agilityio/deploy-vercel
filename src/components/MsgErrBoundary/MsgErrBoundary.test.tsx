import { fireEvent, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import MsgErrBoundary from './MsgErrBoundary';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
}));

const mockNavigate = jest.fn();
jest.spyOn(require('react-router-dom'), 'useNavigate').mockReturnValue(mockNavigate);

it('renders MsgErrBoundary component correctly', () => {
  const { container, getByText } = render(
    <MemoryRouter>
      <MsgErrBoundary />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();

  const backButton = getByText('Back to home page');
  fireEvent.click(backButton);
  expect(mockNavigate).toHaveBeenCalledWith('/');
  expect(mockNavigate).toHaveBeenCalledTimes(1);
});
