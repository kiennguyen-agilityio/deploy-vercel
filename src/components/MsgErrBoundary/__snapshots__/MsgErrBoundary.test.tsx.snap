// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MsgErrBoundary component correctly 1`] = `
<div>
  <div
    class="modal-overlay"
  >
    <div
      class="modal-content"
    >
      <div
        class="modal-header"
      >
        <button
          class="modal-close-button"
          type="button"
        >
          Ã—
        </button>
      </div>
      <div
        class="modal-body"
      >
        <div
          class="error-fallback-container"
        >
          <h2
            class="error-fallback-title"
          >
            Oops! Something went wrong
          </h2>
          <p
            class="error-fallback-description"
          >
            We're sorry, but an error occurred while processing your request.
          </p>
          <button
            class="btn btn-primary btn-color-success btn-sm  "
            type="button"
          >
            <span>
              Back to home page
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
